client/src/js/subtraction/components/Detail.js

```diff
@@ -86,18 +108,32 @@ export class SubtractionDetail extends React.Component {
                         </tr>
                     </tbody>
                 </Table>
-
+                {hasFiles && (
+                    <BoxGroup>
+                        <BoxGroupHeader>
+                            <h2>Files</h2>
+                            <p>Data files available to workflows using this subtraction</p>
+                        </BoxGroupHeader>
+                        {this.props.files.map(file => (
+                            <StyledBoxGroupSection key={file.id}>
+                                <a href={file.download_url}>{file.name}</a>
+                                <strong>{byteSize(file.size)}</strong>
+                            </StyledBoxGroupSection>
+                        ))}
+                    </BoxGroup>
+                )}
```

Comment:
Break this up into modules.
1. Add module with connected component for listing subtraction files.
2. Add module with list item component only.
3. Make a new directory at `subtraction/components/Detail` and put all detail related component modules in here. Relevant tests also have to move.